components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1861,6 +1873,32 @@ private Condition getCondition(Node node, Map<String, String> attributes) throws
             String attributeName = ((ExpressionNode) node).getAttributeValue();
             String attributeValue = ((ExpressionNode) node).getValue();
 
+            try {
+                /*If primary login identifier feature is enabled, the username uri should be replaced with
+                appropriate scim attribute of the primary login identifier claim
+                 */
+                if (SCIMConstants.UserSchemaConstants.USER_NAME_URI.equals(attributeName) &&
+                        isLoginIdentifiersEnabled() && StringUtils.isNotBlank(getPrimaryLoginIdentifierClaim())) {
+                    attributeName = getScimUriForPrimaryLoginIdentifier(node);
+                }
+            } catch (org.wso2.carbon.user.core.UserStoreException e) {
+                throw new CharonException("Error in retrieving scim to local mappings.", e);
+            }
+
+
+            try {
```

Comment:
Is this logic redundant? the same logic is implemented inline 1876? WDYT?

Reply:
updated